### ISSUE TITLE
GroupSpeedDial seems to cause sites to reset to 100% zoom

### ISSUE THREAD

**Issue by DaveyBoy1970** [fastaddons/GroupSpeedDial#19]
Firstly I adore GroupSpeedDial, which is why I'm finding this particular issue painful.

A couple of weeks ago I noticed sites were randomly resetting themselves from the zoom i set them at (typically eith 120% or 133%) back to 100%. I also noticed that the zoomed thumbnails thumbnails I had set on the dials page were also resetting themselves to 100%. I disabled GroupSpeedDial for a week and the problem went away. I missed it a lot so I thought I would go back in and delete the dials and recreate them. After a lot of fiddling I was able to recreate the thumbnails with the correct magnification, and page zoom seemed to be stable. After a day or so the same problem had returned, thumbnails resetting themselves to show the entire page and youtube/facebook/BBC etc resetting themselves back to 100% zoom. I'm not sure but the background refresh may have something to do with it?

What can I do to remedy this?

Many thanks for an otherwise wonderful addon.

**Comment by fastaddons** [comment]
Hello,
I'm really sorry for breaking your zoom!
My extension is indeed changing zoom level when it's making a thumbnail for a page. If you have auto-reload enabled for some pages, it may reset those. But it should revert it back to the previous value afterwards.

Few questions:
1) so this was not happening before and stared only recently? I've released new version 15 recently where I did a lot of changes, including thumbnails generation so maybe I broke something.
2) are you sure the zoom level is always reset to exactly 100%?
3) is it possible that also pages that were not auto-reloaded got reset zoom?

**Comment by DaveyBoy1970** [comment]
Thanks so responding so quickly. To answer your questions:

1. Only recently. It started happening just before Firefox version 80 was released. When it started happening I recall checking to see if there was an upgrade and saw the new version (I'm on Ubuntu) so I upgraded but that didn't change anything, which is when I started checking the addons for problems. Before this I was very happy having a thumbnail zoomed to get the top news story from BBC News and the first two days of my local weather - both of which I have reloading every few hours. 
2. Yes, the zoom is always set back to 100% - I presumed that was because it's the default, but maybe not. I've just set the Firefox default zoom to 120% to see if that makes any difference.
3. Yes. I have YouTube as a dial, but I've just given it a static logo image and it's not set to auto-reload at all, yet that very often jumps back to 100% zoom, sometimes right in front of my eyes.

I also have NoScript installed and I noticed that when I tried to refresh the thumbnails NoScript crashed and I had to completely restart the browser - that may be unrelated though.

**Comment by Juraj-Masiar** [comment]
Thank you for the info! I think I found the problem. The zoom reset should be applied only to the hidden reloading tab, but it was being applied to currently focused tab.

I have few more things to fix so I will be releasing this in 15.2 version in a few days (probably by the end of this week).

**Comment by DaveyBoy1970** [comment]
Oh. That's brilliant. I shall be looking forward to it!

**Comment by fastaddons** [comment]
The 15.2 was released, please let me know if you find more issues.
Thanks and have a nice weekend :)

**Comment by DaveyBoy1970** [comment]
Thank you! Sorry I had to go away for a few days and I couldn't check properly. I've been checking this morning and the zoom problem doesn't seem to be happening so far. However, I'm still having problems with the thumbnails, and I just got this error:

Error getting thumbnails: TypeError: can't convert undefined to object
,
Environment: Group Speed Dial 15.2; info: os: linux, arch: x86-64; name: Firefox, vendor: Mozilla, version: 80.0, buildID: 20200818235255;
Pixel ratio: 1
Running time: 82 / 28050.936
Stats: 0 - 9,3,2
Runtime errors:
undefined
Version: 15.2; OS: {"os":"linux","arch":"x86-64"}; Browser: {"name":"Firefox","vendor":"Mozilla","version":"80.0","buildID":"20200818235255"};
